src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -223,7 +223,7 @@ private void UnregisterGlobalOperationNotifications()
                 // Unilaterally transition us to the finished state.  Once we're finished
                 // we cannot start or stop anymore.
                 _globalNotificationsTask = _globalNotificationsTask.ContinueWith(
-                    _ => GlobalNotificationState.Finished, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
+                    _ => GlobalNotificationState.Finished, _shutdownCancellationTokenSource.Token, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
So is UnregisterGlobalOperationNotification only called in the one Dispose() case? So then why don't we just get rid of the Wait() call entirely? Otherwise this looks like we're setting a cancellation to queue a task that we'll then not wait for because it was cancelled?